### ISSUE TITLE
Effect sizes?

### ISSUE THREAD

**Issue by scroono** [gamlj/gamlj#183]
Hi! I love this module for teaching my undergraduate psychology students - it gives them easy access to a whole world of advanced analyses. One thing that we have a barrier with is getting some measure of magnitude of effect for the fixed factors in the analyses. Would it be possible to implement something to assist with this in future versions? I realise that this is not a simple process!

**Comment by mcfanda** [comment]
Yes, we are thinking about the best way to proceed about this issue. The literature does not help much, but  I agree that something should be done

**Comment by rayabbasi8** [comment]
Hi when would this be possible by any chance? Have a masters due on the 10th and it really would help me a lot

**Comment by mcfanda** [comment]
Surely not that fast. The reason is that there is no consensus on how to compute them. We do not want to implement something that it is not consolidated in the literature.

**Comment by rayabbasi8** [comment]
Understood, would you have any references that say this so I may be able to add that to my analysis?

**Comment by mcfanda** [comment]
well, it is the lack of references that creates the issue. There are some references proposing some sort of effect size, but they are not convincing yet.

**Comment by rayabbasi8** [comment]
I see, what would be the best way to go about this then? 
there are many papers in my field that have done this with LMM but are their findings relevant then?

**Comment by mcfanda** [comment]
please, send the references by email, but I am afraid they are the ones I am referring to

**Comment by chantelanuit** [comment]
Hello,

What about the **upsilon effect size suggest by: Lachowicz, M. J., Preacher, K. J., and Kelley, K. (2018)**. A novel measure of effect size for mediation analysis. Psychol. Method. 23, 244–261. doi: 10.1037/met0000165

Upsilon is available in MBESS https://cran.r-project.org/web/packages/MBESS/MBESS.pdf

"Mediation analysis has enabled behavioral researchers to better understand the mechanistic relationships between variables. Many researchers are specifically interested in the role of mediators (M) that account for the relation between a predictor variable (X; independent variable) and an outcome variable (Y; dependent variable). A variety of effect size measures have been developed for mediation analysis. However, most of these effect size measures have limitations, including non-monotonicity and spurious inflation. [Lachowicz et al. (2018)](https://www.frontiersin.org/journals/psychology/articles/10.3389/fpsyg.2023.1101440/full#B13) developed a new effect size measure, upsilon (υ), which has overcome these two limitations. The current study extends its work by applying this effect size metric to mediation models with a multicategorical predictor." cited in https://www.frontiersin.org/journals/psychology/articles/10.3389/fpsyg.2023.1101440/full#B13

**Comment by mcfanda** [comment]
That pertains mediation, GAMLj does not really deal with mediation

**Comment by chantelanuit** [comment]
sorry, I posted it on the wrong post, I was referring to jamm

**Comment by mcfanda** [comment]
sure, we can do that (but upsilon is simply the square of "beta" that you find already in the output)

**Comment by chantelanuit** [comment]
I was referring to this:

![upsilon](https://github.com/gamlj/gamlj/assets/24530165/19d67898-309b-4a7c-afcc-67e984471c8e)

**Comment by sjentsch** [comment]
The following article https://link.springer.com/article/10.1007/s40300-025-00295-w and the code in this repo might help: https://github.com/AMoeller11/LMMCohensf2 (the repo is currently lacking the data file, but I asked for it to be added).
